projects/graph/src/draw.py

```diff
@@ -1,14 +1,74 @@
 """
 General drawing methods for graphs using Bokeh.
 """
-
+import math
 from bokeh.io import show, output_file
 from bokeh.plotting import figure
-from bokeh.models import (GraphRenderer, StaticLayoutProvider, Circle, LabelSet,
+from bokeh.models import (GraphRenderer, StaticLayoutProvider, Oval, Circle, LabelSet,
                           ColumnDataSource)
-
+from bokeh.palettes import Spectral8
+from graph import Graph
 
 class BokehGraph:
     """Class that takes a graph and exposes drawing methods."""
-    def __init__(self):
-        pass  # TODO
+    def __init__(self, Graph):
+        self.Graph = Graph
+    def draw(self):
+        Graph = self.Graph
+
+N = 8
```

Comment:
this code is used for demo purposes, you will need to modify it so that bokeh will draw from the graph class you defined. Please take a look at https://github.com/br80/Graphs-CS12/blob/master/projects/graph/src/draw.py for reference.